### DISCUSSION TITLE
Cannot connect PnP Search result to a search filter web part.  "No available filter to display" is shown

### DISCUSSION THREAD

**Discussion by shirleysham** [microsoft-search/pnp-modern-search#2748] [Q&A]
I have a search results web part, search box, and a search filter.  I have configured the search filter to connect to the search results, and vice versa.  I have added the filter in customize filters.  However I'm still getting "No available filter to display" when saving the filter.

I also have a second question, is it possible to use the filter in a deep link in URL? Many thanks

**Comment by shirleysham** [comment]
![image](https://user-images.githubusercontent.com/121912343/217148892-58bdb192-2391-43f0-9f4f-fdc13e12c760.png)
![image](https://user-images.githubusercontent.com/121912343/217148947-2441cd6f-f317-40c6-b485-0435b7cb0259.png)
![image](https://user-images.githubusercontent.com/121912343/217149018-c2c17d56-7e3a-499b-b6b3-bad4d360adde.png)

**Comment by kasperbolarsen** [comment]
Hi @shirleysham 

Which version are you using?
What does you Query Template looks like?
Is the field you intend to use as a filter Refinable?
 Is the field added to Selected Properties?

**Comment by shirleysham** [comment]
Hello Kasper,

Much appreciate your response! Please see my replies below.

Which version are you using?  Version: 4.8.0

What does you Query Template looks like?
{searchTerms} path:
https://kwmhkg.sharepoint.com/sites/HK/Lists/HK%20Finance%20Accordian

Is the field you intend to use as a filter Refinable? I don't think it's
refinable, as you can see from sharepoint admin (managed properties)
below.  May I know how to make it refinable? It's dimmed in Sharepoint
admin.
[image: image.png]


Is the field added to Selected Properties?
Yes

Thanks a million.

**Reply by wobba** [reply]
You may only use managed properties in the search schema which are set to be "Refinable" as a filer. E.g. the re-usable ones RefinableStringXXX properties.

**Comment by kasperbolarsen** [comment]
It seems like the image of the Managed property isn't uploaded. I'll assume it is a custom site column named Category or something like that.  The field is most likely a Managed Metadata field. 
If this assumption is correct I would map ows_Category to a RefinableString, wait 24 hours or until I see data in the RefinableString, add the RefinableString to the Selected Properties and finally add it a refiner 😁

**Reply by shirleysham** [reply]
Thanks both I've just updated RefinableString00 and mapped it to a crawled property.  Also reindexed the field in the SharePoint list.  Now waiting 24 hours and see if I can use it in the Search filter.  
Thanks.

![image](https://user-images.githubusercontent.com/121912343/217190116-a85b9566-24ae-4b82-aa8b-626eeb06efe8.png)
![image](https://user-images.githubusercontent.com/121912343/217190215-7230caaf-fdb1-48dd-8dac-92c8b523d899.png)

**Reply by kasperbolarsen** [reply]
Is the internal name of the site column really o8hj ?

**Reply by shirleysham** [reply]
Yes it's o8hj

**Comment by shirleysham** [comment]
Hi Kasper,

Firstly thanks for your advice to use a refinable managed property for the
search filter.  The filter can now be added to the page and I can see the
filter in the URL when selecting a value in the filter, ie.

https://XXXhkg.sharepoint.com/sites/HK/SitePages/people11.aspx?f=%5B%7B%22filterName%22%3A%22RefinableString00%22%2C%22values%22%3A%5B%7B%22name%22%3A%22Disbursement%2FExpense%22%2C%22value%22%3A%22%5C%22%C7%82%C7%8244697362757273656d656e742f457870656e7365%5C%22%22%2C%22operator%22%3A0%7D%5D%2C%22operator%22%3A%22or%22%7D%5D#

This translates to something like this when decoding the URL:

https://XXXhkg.sharepoint.com/sites/HK/SitePages/people11.aspx?f=[{
"filterName":"RefinableString00","values":[{"name":"Disbursement/Expense","value":"\"ǂǂ44697362757273656d656e742f457870656e7365\"","operator":0}],"operator":"or"}]

I would like to generate this string myself.  I would like to know what the
value "44697362757273656d656e742f457870656e7365" is corresponding to and
how I can get this value? Is there another way to filter in the URL without
this value? Appreciate your help.  Thanks.
[image: image.png]

**Comment by shirleysham** [comment]
I came across this thread 
https://github.com/microsoft-search/pnp-modern-search/issues/385

Can I add something like the below to filter the results in the URL:
?Filters=[{"n":"RefinableString00","t":["string('Disbursement/Expense')"]}]

What config do I need in the search results to do this? Thanks again....

**Comment by shirleysham** [comment]
No worries, I've solved the problem by using an initial refinement filter in the search results.  Thanks.

RefinableString00:equals("{QueryString.Cat}")

**Reply by wobba** [reply]
Or you could set the same in the query template field as: `RefinableString00:{QueryString.Cat}`
